src/machine.rs

```diff
@@ -0,0 +1,148 @@
+use std::collections::HashMap;
+use std::mem::align_of;
+
+use rustc::mir::{Mir, Local};
+
+use interp::TypedVal;
+
+const X64_ALIGN_SIZE: u8 = 8;
+const X32_ALIGN_SIZE: u8 = 4;
+
+#[derive(Debug)]
+struct Frame<'tcx> {
+    // MIR is a CFG of a single function in rust, so it makes sense
+    // for this to be in the frame.
+    mir: Mir<'tcx>,
+
+    // FIXME: Hacky... I can't find a way to actually access the private u32
+    // inside the `Local` tuple struct. `Local` does  implement the Hashable
+    // and PartialEQ trait however, so for  now it might just work to to use it
+    // as the key
```

Comment:
We really need access to that number, because otherwise we're going to be stuck with a (very slow) `HashMap` rather than a (very fast) array. [Plus if you have numbers, there are other tricks you can do even before you think about arrays!]

Reply:
Okay, this is something I shall ask about on the #rustc IRC.

Reply:
Raise an issue so it's not forgotten?